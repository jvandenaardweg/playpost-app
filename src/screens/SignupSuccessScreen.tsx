import React from 'react';
import {NavigationScreenProp, NavigationRoute } from 'react-navigation';

import { EmptyState } from '../components/EmptyState';

interface Props {
  navigation: NavigationScreenProp<NavigationRoute>
}

export class SignupSuccessScreen extends React.PureComponent<Props> {
  static navigationOptions = {
    title: 'Signup success',
    header: null
  };

  render() {
    return (
      <EmptyState title="Welcome! You are ready to go" description="You can now add articles to your playlist from every app on your phone." actionButtonLabel='Go to my playlist' actionButtonOnPress={() => this.props.navigation.navigate('Playlist')} />
    );
  }
}
