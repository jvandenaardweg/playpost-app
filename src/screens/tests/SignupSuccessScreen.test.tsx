import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';
import { SignupSuccessScreen } from '../SignupSuccessScreen';

const navigateHandler = jest.fn();
const navigationGetParamHandler = jest.fn();
const navigationGoBackHandler = jest.fn();

const props: any = {
  navigation: {
    navigate: navigateHandler,
    getParam: navigationGetParamHandler,
    goBack: navigationGoBackHandler,
  },
};

describe('SignupSuccessScreen', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      wrapper = render(<SignupSuccessScreen {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
