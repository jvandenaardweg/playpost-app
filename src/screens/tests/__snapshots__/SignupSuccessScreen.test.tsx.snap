// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupSuccessScreen rendering should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EEEEEE",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingLeft": 12,
      "paddingRight": 12,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "600",
          "marginBottom": 6,
          "textAlign": "center",
        }
      }
      testID="empty-state-title"
    >
      Welcome! You are ready to go
    </Text>
    <Text
      style={
        Object {
          "color": "#AAAAAA",
          "fontSize": 15,
          "lineHeight": 24,
          "marginBottom": 14,
          "paddingLeft": 24,
          "paddingRight": 24,
          "textAlign": "center",
        }
      }
      testID="empty-state-description"
    >
      You can now add articles to your playlist from every app on your phone.
    </Text>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="empty-state-button"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 24,
              "padding": 8,
              "paddingLeft": 24,
              "paddingRight": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Go to my playlist
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
